python/ray/actor.py

```diff
@@ -1515,30 +1581,31 @@ def __init__(
                     module_name, method_name, class_name
                 )
                 self._ray_function_descriptor[method_name] = function_descriptor
-                method = ActorMethod(
-                    self,
-                    method_name,
-                    self._ray_method_num_returns[method_name],
-                    self._ray_method_max_task_retries.get(
-                        method_name, self._ray_max_task_retries
-                    )
-                    or 0,  # never None
-                    self._ray_method_retry_exceptions.get(method_name),
-                    self._ray_method_is_generator[method_name],
-                    self._ray_method_generator_backpressure_num_objects.get(
-                        method_name
-                    ),  # noqa
-                    self._ray_method_enable_task_events.get(
-                        method_name,
-                        self._ray_enable_task_events,  # Use actor's default value
-                    ),
-                    decorator=self._ray_method_decorators.get(method_name),
-                    signature=self._ray_method_signatures[method_name],
-                    tensor_transport=self._ray_method_name_to_tensor_transport.get(
-                        method_name
-                    ),
+
+        # Build one ActorMethodShell per method, avoid cycles
```

Comment:
```suggestion
        # Build an ActorMethodShell per method to cache expensive parsing logic.
        # The ActorMethodShell doesn't take a reference to this ActorHandle to avoid a circular reference.
        # Instead, we will lazily bind this ActorHandle to the ActorMethodShell when a method is invoked.
```